js/modules/GLPI.js

```diff
@@ -0,0 +1,209 @@
+        ]);
+    }
+
+    async registerKnownModule(key) {
+        if (this.known_modules[key] !== undefined) {
+            let m = this.known_modules[key];
+            if (typeof m === 'string') {
+                // Prevent directory traversal and other issues (Remove all characters that are not a-zA-Z0-9_- or forward slash)
+                m = m.replace(/[^a-zA-Z\d_\-/]/g, '');
+                // Load the module from a script
+                const { default: m_class } = await import('./' + m + '.js');
+                this.registerModule(key, new m_class());
+            } else {
+                this.registerModule(key, this.known_modules[key]);
+            }
+        } else {
+            throw new Error(`${key} is not a known module.`);
+        }
+    }
+
+    /**
+     * Register a module.
+     * @param {string} key The module key
+     * @param {GLPIModulePlugin} module
+     */
+    async registerModule(key, module) {
+        // Check if the module is already registered
+        if (this.isModuleRegistered(key)) {
+            return;
+        }
+        // Load dependencies
+        const dependencies = module.getDependencies();
+        if (dependencies.length > 0) {
+            dependencies.forEach((dependency) => {
+                if (!this.isModuleRegistered(dependency.key)) {
+                    // Try to register the dependency
+                    try {
+                        this.registerKnownModule(dependency.key);
+                    } catch (e) {
+                        // Log the error
+                        console.error(e);
+                        if (!dependency.optional) {
+                            throw new Error (`Module "${key}" could not be loaded as one or more required dependencies could not be loaded.`);
+                        }
+                    }
+                }
+            });
+        }
+
+        this.modules[key] = module;
+        this.bindLegacyGlobals(module);
+        module.initialize();
+        this.getEventTarget().dispatchEvent(new CustomEvent('module:registered', {
+            detail: {
+                module: module
+            }
+        }));
+    }
+
+    /**
+     * Bind functions to "window" for legacy (non-module) support.
+     * @param {GLPIModulePlugin} module
+     */
+    bindLegacyGlobals(module) {
+        const globals = module.getLegacyGlobals();
+
+        Object.entries(globals).forEach(([key, value]) => {
+            if (window[key] !== undefined) {
+                throw new Error(`Legacy global "${key}" already exists.`);
+            }
+            if (typeof value === 'string') {
+                // Simple global
+                window[key] = module[value];
+            } else {
+                // Advanced global (Allow binding a different context)
+                const bind_target = value.bind_target || window;
+                window[key] = module[value.module_property].bind(bind_target);
+            }
```

Comment:
Did you find a case where using a specific `bind_target` would be usefull ?

I think we should simplify code here to `window[key] = module[value].bind(module);`. If a developper want to bind to something else than the module itself, it would be able to require a dedicated method, like `legacyMethod(params) { this.targetMethod.bind(another_object); }`.

Reply:
Currently, the new dialogs code needs it so their functions that get added to windows have access to the other methods inside the module.

Reply:
Also, here, we should probably introduce a deprecation log, with someting like this:
```js
window[key] = function () {
    module[value].apply(module, Array.from(arguments));
}
```